src/types.jl

```diff
@@ -68,7 +75,7 @@ Important fields:
 struct FrameCode
     scope::Union{Method,Module}
     src::CodeInfo
-    methodtables::Vector{Union{Compiled,TypeMapEntry}} # line-by-line method tables for generic-function :call Exprs
+    methodtables::Vector{Union{Compiled,DispatchableMethod}} # line-by-line method tables for generic-function :call Exprs
```

Comment:
I don't see how this prevents `which`-based lookup on each iteration of a loop. In #309 I added a `compiled` field to `DispatchableMethod`, which seems like the right place for this information (once you've determined that a method matches, your next question is whether you should run it in interpreted or compiled mode).

Reply:
It prevents it because `frameinstance` in `DispatchableMethod`is a `Compiled` in those cases.

Reply:
The reason I still want to have `Compiled` directly in `methodtables` is for ccall / fcall where we know for sure that the method we want to call are compiled so there is no need to do any matching against signatures.